### PR TITLE
doc: add grpc proxy doc

### PR DESCRIPTION
**Pull request by xiang90**
/cc @gyuho @heyitsanthony

### CODE REVIEW

File: Documentation/op-guide/grpc_proxy.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            +-------------+
+            | etcd server |
+            +------+------+
+                   ^ watch key A
+                   |
+           +-------+-----+
+           | gRPC proxy  | <-------+
+           |             |         |
+           ++-----+------+         |watch key A
+watch key A ^     ^ watch key A    |
```
Comment (heyitsanthony) [inline Documentation/op-guide/grpc_proxy.md]: indicate these are c-watchers?

File: Documentation/op-guide/grpc_proxy.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Assuming N clients watch the same key, one gRPC proxy can reduce the watch load on the etcd server from N to 1. Users can deploy multiple gRPC proxies to further distribute server load.
+
+In the following example, three clients watch on key A. The gRPC proxy coalesces the three watchers, creating a single  watcher attached to the etcd server.
+
+```
+            +-------------+
+            | etcd server |
+            +------+------+
+                   ^ watch key A
+                   |
```
Comment (heyitsanthony) [inline Documentation/op-guide/grpc_proxy.md]: s-watcher?

File: Documentation/op-guide/grpc_proxy.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            |     |                |
+    +-------+-+  ++--------+  +----+----+
+    |  client |  |  client |  |  client |
+    |         |  |         |  |         |
+    +---------+  +---------+  +---------+
+```
+
+### Limitations
+
+To effectively coalesce multiple client watchers into a single watcher, the gRPC proxy coalesces new client watchers into an existing watcher when possible. This coalesced watcher may be out of sync with the etcd server due to network delays or buffered undelivered events. When the watch revision is unspecified, the gRPC proxy will not guarantee the client watcher will start watching from the most recent store revision. For example, if the client watches from an etcd server with revision 1000, that watcher will begin at revision 1000. If the client watches from the gRPC proxy, may begin watching from revision 990.
```
Comment (heyitsanthony) [inline Documentation/op-guide/grpc_proxy.md]: Should this use the c-watcher / s-watcher terminology?
`
To effectively coalesce multiple client watchers into a single watcher, the gRPC proxy coalesces new c-watchers into an existing s-watcher when possible. This coalesced s-watcher may be out of sync with the etcd server due to network delays or buffered undelivered events. When the watch revision is unspecified, the gRPC proxy will not guarantee the c-watcher will start watching from the most recent store revision. For example, if a client watches from an etcd server with revision 1000, that watcher will begin at revision 1000. If a client watches from the gRPC proxy, may begin watching from revision 990.`

File: Documentation/op-guide/grpc_proxy.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+Similar limitations apply to cancellation. When the watcher is cancelled, the etcd server’s revision may be greater than the cancellation response revision. 
+
+These two limitations should not cause problems for most use cases. In the future, there may be add options to force the watcher to bypass the gRPC proxy for more accurate revision responses.
```
Comment (heyitsanthony) [inline Documentation/op-guide/grpc_proxy.md]: s/add/additional

File: Documentation/op-guide/grpc_proxy.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+### Limitations
+
+To effectively coalesce multiple client watchers into a single watcher, the gRPC proxy coalesces new client watchers into an existing watcher when possible. This coalesced watcher may be out of sync with the etcd server due to network delays or buffered undelivered events. When the watch revision is unspecified, the gRPC proxy will not guarantee the client watcher will start watching from the most recent store revision. For example, if the client watches from an etcd server with revision 1000, that watcher will begin at revision 1000. If the client watches from the gRPC proxy, may begin watching from revision 990.
+
+Similar limitations apply to cancellation. When the watcher is cancelled, the etcd server’s revision may be greater than the cancellation response revision. 
+
+These two limitations should not cause problems for most use cases. In the future, there may be add options to force the watcher to bypass the gRPC proxy for more accurate revision responses. 
+
+## A scalable lease API
```
Comment (heyitsanthony) [inline Documentation/op-guide/grpc_proxy.md]: `## Scalable lease API`?

**Comment by xiang90** [comment]
@heyitsanthony All fixed. PTAL.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**